transport/http_util.go

```diff
@@ -127,6 +127,17 @@ func isReservedHeader(hdr string) bool {
 	}
 }
 
+// isWhitelistedHttp2Header checks whether hdr belongs to HTTP2 headers
+// that should be propagated into metadata visible to users.
+func isWhitelistedHttp2Header(hdr string) bool {
+	switch hdr {
+	case ":authority", "Authority":
```

Comment:
Indeed, I thought this would be needed for http2 Handler implementation, but that required additional code anyway.